### DISCUSSION TITLE
Adding Mixpanel analytics to Remix app (Server-side + Client-side)

### DISCUSSION THREAD

**Discussion by wengzilla** [remix-run/remix#5079] [Q&A]
Does anyone have any examples or suggestions on how I would implement Mixpanel analytics in Remix?

Mixpanel has two separate libraries: A [Client-side library](https://developer.mixpanel.com/docs/javascript-quickstart) and a [Server-side library](https://developer.mixpanel.com/docs/nodejs).

Some questions:
- Does Remix have some magic in it whereby I could get away with only using one of the libraries or will I need to use both?
- On the server-side, my understanding is that it's generally recommended to create a global instance of the tracking library (I think to handle buffering events and whatnot). Does it make sense to export a const variable from the entry.server.tsx file and then import it in the respective `action` and `loader` functions?

Thanks!

**Comment by Dev-Fouad** [comment]
hi @wengzilla , where you able to fix this? I need guide on how to go about it.
